device-connectors/src/testflinger_device_connectors/devices/zapper_iot/__init__.py

```diff
@@ -13,22 +13,84 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 """Zapper Connector for IOT provisioning."""
-
-from typing import Any, Dict, Tuple
-
+import logging
+import json
+import yaml
 from testflinger_device_connectors.devices.zapper import ZapperConnector
+from testflinger_device_connectors.devices.zapper_iot.parser import (
+    validate_data,
+    validate_url,
+)
+import rpyc
+
+logger = logging.getLogger(__name__)
 
 
 class DeviceConnector(ZapperConnector):
     """Tool for provisioning baremetal with a given image."""
 
     PROVISION_METHOD = "ProvisioningIoT"
+    ZAPPER_REQUEST_TIMEOUT = 60 * 70
+    ZAPPER_SERVICE_PORT = 60000
+
+    def __init__(self):
+        self.config = None
+        self.job_data = None
+
+    def provision(self, args):
+        """Method called when the command is invoked."""
+        with open(args.config, encoding="utf-8") as configfile:
+            self.config = yaml.safe_load(configfile)
+        with open(args.job_data, encoding="utf-8") as job_json:
+            self.job_data = json.load(job_json)
 
-    def _validate_configuration(
-        self,
-    ) -> Tuple[Tuple, Dict[str, Any]]:
```

Comment:
is there any reason why typing hints were removed?

Reply:
Sorry I didn't notice that, will fix it later.

Reply:
@nadzyah Hi Nadzyah
I have updated the changes, could you help to review is there any inappropriate 
Thank you